account_customer_wallet/tests/test_balance.py

```diff
@@ -101,3 +101,21 @@ def test_debit_and_credit(self):
         self._create_move(debit=50)
 
         self.assertEqual(self.partner.customer_wallet_balance, 50)
+
+    def test_no_wallet_account(self):
+        """If no wallet account is set, expect the global toggle to be turned off,
+        and other functionality to not crash and break.
+        """
+        self._create_move(credit=100)
+
+        self.env["ir.config_parameter"].set_param(
+            "account_customer_wallet.customer_wallet_account_id",
+            None,
+        )
+        self.assertFalse(
+            self.env["ir.config_parameter"].get_param(
+                "account_customer_wallet.customer_wallet"
+            )
+        )
```

Comment:
Not sure why the test fails here. It's like `_onchange_customer_wallet` in `res.config.settings` is never triggered. As a matter of fact, it isn't—I tested in the debugger.

Reply:
`res.config.settings` is a TransientModel with helper functions, some of which use `set_param` or `get_param` to write in `ir.config_parameter`.

So here, no `res.config.settings` is created for the onchange to be triggered.

More over : the onchange functions are only triggered when **forms** are modified. In tests, you need to use the `with Form(some object):` to test that behavior.

All of this is probably obsolete if you set the required constraint on the form as I suggested earlier.